File: java/src/sdk/dataobjects/paypal/payflow/Devices.java
package paypal.payflow;

/**
 * Used for the type of device the card holder is using.
 * .................
 * //inv is the Invoice object.
 * .................
 *
 * //Set the Shipping Address details.
 * ShipTo ship = new ShipTo();
 * ship.setShipToStreet( "123 Main St.");
 * ship.setShipToZip("12345");
 * inv.setShipTo (ship);
 * .................
 */

public final class Devices extends BaseRequestDataObject {
    private String catType;
    private String contactLess;

    /**
     * Gets the type of terminal
     *
     * @return String
     * <p>Maps to Payflow Parameter: CATTYPE</p>
     */
    public String getCatType() {
        return catType;
    }

    /**
     * Sets the type of terminal
     *
     * @param catType String
     * <p>Maps to Payflow Parameter: CATTYPE</p>
     */
    public void setCatType(String catType) {
        this.catType = catType;
    }

    /**
     * Gets the card input capability
     *
     * @return contactLess String
     *  <p>Maps to Payflow Parameter: CONTACTLESS</p>
     */

    public String getContactLess() {
        return contactLess;
    }

    /**
     * Sets the card input capability
     *
     * @param contactLess String
     *  <p>Maps to Payflow Parameter: CONTACTLESS</p>
     */
    public void setContactLess(String contactLess) {
        this.contactLess = contactLess;
    }

    /**
     * Used for the type of device the card holder is using.
     * .................
     * //inv is the Invoice object.
     * .................
     *
     * //Set the Shipping Address details.
     * ShipTo ship = new ShipTo();
     * ship.setShipToStreet( "123 Main St.");
     * ship.setShipToZip("12345");
     * inv.setShipTo (ship);
     * .................
     */

    public Devices() {
    }

    protected void generateRequest() {
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_CATTYPE, catType));
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_CONTACTLESS, contactLess));
    }

}

File: dotNET/PFProSDK/src/PayPal/Payments/DataObjects/Devices.cs
#region "Imports"

using System;
using PayPal.Payments.Common.Utility;
using PayPal.Payments.Common.Exceptions;

#endregion

namespace PayPal.Payments.DataObjects
{	
		/// <summary>
		/// Constructor
		/// </summary>
		/// <remarks>Items that reflect what type of device; either terminal or card is used or presented.</remarks>
		/// <example>
		/// <para>Following example shows how to use Devices.</para>
		/// <code lang="C#" escaped="false">
		///  .................
		///  //Inv is the Invoice object.
		///  .................
		/// // Create a new Devices object.
		///	Devices UsedDevices = new Devices();
		///	UsedDevices.CatType = "3";
		///	UsedDevices.Contactless = "RFD";
		///	Inv.Devices = UsedDevices;
		///	.................
		/// </code>
		/// <code lang="Visual Basic" escaped="false">
		/// .................
		/// 'Inv is the Invoice object.
		/// .................
        /// ' Set the device/card capabilities
        /// Dim UsedDevices As Devices = New Devices
        /// UsedDevices.CatType = "3"
        /// UsedDevices.Contactless = "RFD"
        /// Inv.Devices = UsedDevices
		///	.................
		/// </code>
		/// </example>

    public sealed class Devices : BaseRequestDataObject
    {
        #region "Member Variables"

		/// <summary>
		/// Type of Terminal
		/// </summary>
		private String mCatType;

		/// <summary>
		/// Card Input Capability
		/// </summary>
		private String mContactless;

		#endregion

		#region "Properties"

		/// <summary>
		/// Gets, Sets Terminal type
		/// </summary>
		/// <remarks>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>CATTYPE</code>
		/// </remarks>
		public String CatType
		{
            get { return mCatType; }
            set { mCatType = value; }
        }

		/// <summary>
		/// Gets, Sets Card Input Capability
		/// </summary>
		/// <remarks>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>CONTACTLESS</code>
		/// </remarks>
		public String Contactless
		{
            get { return mContactless; }
            set { mContactless = value; }
		}

		#endregion

		#region "Constructors"

		/// <summary>
		/// Constructor
		/// </summary>
		/// <remarks>Items that reflect what type of device; either terminal or card is used or presented.</remarks>
		/// <example>
		/// <para>Following example shows how to use Devices.</para>
		/// <code lang="C#" escaped="false">
		///  .................
		///  //Inv is the Invoice object.
		///  .................
		/// // Create a new Devices object.
		///	Devices UsedDevices = new Devices();
		///	UsedDevices.CatType = "3";
		///	UsedDevices.Contactless = "RFD";
		///	Inv.Devices = UsedDevices;
		///	.................
		/// </code>
		/// <code lang="Visual Basic" escaped="false">
		/// .................
		/// 'Inv is the Invoice object.
		/// .................
        /// ' Set the device/card capabilities
        /// Dim UsedDevices As Devices = New Devices
        /// UsedDevices.CatType = "3"
        /// UsedDevices.Contactless = "RFD"
        /// Inv.Devices = UsedDevices
		///	.................
		/// </code>
		/// </example>
		public Devices()
		{
		}

		#endregion

		#region "Core functions"

		/// <summary>
		/// Generates the transaction request.
		/// </summary>
		internal override void GenerateRequest()
		{
			try
			{
				RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_CATTYPE, mCatType));
				RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_CONTACTLESS, mContactless));
		
			}
			catch (BaseException)
			{
				throw;
			}
			catch (Exception Ex)
			{
				DataObjectException DEx = new DataObjectException(Ex);
				throw DEx;
			}
		}

		#endregion
	}
}